```
[00:11:13]           │      Caused by: org.elasticsearch.action.NoShardAvailableActionException: [ftr][127.0.0.1:9300][indices:data/read/search[phase/query]]
[00:11:13]           └- ✖ fail: Fleet Endpoints Integrations inputs_with_standalone_docker_agent generate a valid config for standalone agents
[00:11:13]           │      ResponseError: search_phase_execution_exception
[00:11:13]           │ 	Root causes:
[00:11:13]           │ 		no_shard_available_action_exception: [ftr][127.0.0.1:9300][indices:data/read/search[phase/query]]
[00:11:13]           │       at SniffingTransport._request (node_modules/@elastic/transport/src/Transport.ts:601:17)
[00:11:13]           │       at processTicksAndRejections (node:internal/process/task_queues:95:5)
[00:11:13]           │       at /opt/buildkite-agent/builds/bk-agent-prod-gcp-1743506118202442333/elastic/kibana-flaky-test-suite-runner/kibana/node_modules/@elastic/transport/src/Transport.ts:704:22
[00:11:13]           │       at SniffingTransport.request (node_modules/@elastic/transport/src/Transport.ts:701:14)
[00:11:13]           │       at Client.SearchApi [as search] (node_modules/@elastic/elasticsearch/src/api/api/search.ts:96:10)
[00:11:13]           │       at Context.<anonymous> (inputs_with_standalone_docker_agent.ts:79:27)
```

--- a/x-pack/test/fleet_api_integration/apis/integrations/inputs_with_standalone_docker_agent.ts
+++ b/x-pack/test/fleet_api_integration/apis/integrations/inputs_with_standalone_docker_agent.ts
@@ -82,7 +82,7 @@ ${inputsYaml}
             q: `agent.name:${agent.name}`,
             ignore_unavailable: true,
           });
-  
+
           // @ts-expect-error TotalHit
           if (searchRes.hits.total.value > 0) {
             foundMetrics = true;
@@ -90,7 +90,7 @@ ${inputsYaml}
           }
         } catch (err) {
           log.error(err);
-        }  
+        }
         // await agent.log(); uncomment if you need to debug agent logs
         await new Promise((resolve) => setTimeout(resolve, 5 * 1000));
       }

--- a/x-pack/test/fleet_api_integration/apis/integrations/inputs_with_standalone_docker_agent.ts
+++ b/x-pack/test/fleet_api_integration/apis/integrations/inputs_with_standalone_docker_agent.ts
@@ -76,17 +76,21 @@ ${inputsYaml}
       const MAX_ITERATIONS = 20;
       let foundMetrics = false;
       for (let i = 0; i < MAX_ITERATIONS; i++) {
-        const searchRes = await es.search({
-          index: 'metrics-system.cpu-default',
-          q: `agent.name:${agent.name}`,
-          ignore_unavailable: true,
-        });
-
-        // @ts-expect-error TotalHit
-        if (searchRes.hits.total.value > 0) {
-          foundMetrics = true;
-          break;
-        }
+        try {
+          const searchRes = await es.search({
+            index: 'metrics-system.cpu-default',
+            q: `agent.name:${agent.name}`,
+            ignore_unavailable: true,
+          });
+  
+          // @ts-expect-error TotalHit
+          if (searchRes.hits.total.value > 0) {
+            foundMetrics = true;
+            break;
+          }
+        } catch (err) {
+          log.error(err);
+        }  
         // await agent.log(); uncomment if you need to debug agent logs
         await new Promise((resolve) => setTimeout(resolve, 5 * 1000));
       }

